libs/luna-empire/src/Empire/Commands/Typecheck.hs

```
@@ -68,35 +84,37 @@ runInterpreter path imports = runASTOp $ do
         IR.ASGFunction _ [] b -> do
             bodyRef <- IR.source b
             res     <- Interpreter.interpret' imports . IR.unsafeGeneralize $ bodyRef
-            result  <- liftIO $ withCurrentDirectory (maybe (takeDirectory path) Path.toFilePath rootPath) $ runIO $ runError $ execStateT res def
-            case result of
-                Left e  -> return Nothing
-                Right r -> return $ Just r
+            mask_ $ do
+                result  <- liftIO $ withCurrentDirectory (maybe (takeDirectory path) Path.toFilePath rootPath) $ runIO $ runError $ execStateT res def
+                case result of
+                    Left e  -> return Nothing
+                    Right r -> return $ Just r
```

`return $ either Nothing (Just r) result` ?

---

@kustosz I see we're ignoring the exception string here, was it done on purpose?

---

It was done because I had no better idea what to do with this error. This says "the whole computation ended with an exception", without even a hint as to what node caused it. At some point it will be probably displayed somewhere, but is ignored for a long time now.